app/helpers/profile_helper.rb

```diff
@@ -0,0 +1,16 @@
+module ProfileHelper
+  def social_authentication_links_for(user)
+    # Returns a Hash containing URLs for social authentication providers
+    # Currently supported: GitHub, Twitter, Facebook
+    user_identities = user.identities_enabled
+    return {} if user_identities.blank?
+
+    urls = user_identities.pluck(:auth_data_dump).each_with_object({}) do |data, hash|
```

Comment:
Are identies_enabled included in the `@user` object passed to the view, or is this a second database request?